python/tank/log.py

```diff
@@ -100,69 +212,96 @@ def get_root_logger(cls):
+        When you start an engine via the :meth:`sgtk.platform.start_engine` method,
+        a file handler will automatically be created if one doesn't already exist.
 
         If you are manually launching toolkit, we recommend that you call
-        this method to initialize logging to file. Calling it multiple times
-        will not result in the information being written to multiple different
-        files - only one file logger can exist per Toolkit session.
+        this method to initialize logging to file early on in your setup.
+        Calling it multiple times will not result in the information being
+        written to multiple different files - only one file logger can
+        exist per session.
 
         :param log_name: Name of logger to create. This will form the
-                         filename of the log file.
+                         filename of the log file. If you pass None, you
+                         ensure that any existing base file handlers terminate
+                         their logging.
+        :returns: The name of the previous log_name that is being switched away from.
         """
-        # avoid cyclic references
-        from .util import LocalFileStorageManager
-        from .util import filesystem
 
-        if cls._std_file_handler:
+        previous_log_name = self._std_file_handler_log_name
+        log.debug("Switching file based std logger from %s to %s" % (previous_log_name, log_name))
+
+        if self._std_file_handler:
             # there is already a log handler.
             # terminate previous one
-            sgtk_root_logger.removeHandler(cls._std_file_handler)
-            cls._std_file_handler = None
+            log.debug(
+                "Tearing down existing log handler '%s' (%s)" % (previous_log_name, self._std_file_handler)
+            )
+            self._root_logger.removeHandler(self._std_file_handler)
+            self._std_file_handler = None
 
-        # set up logging root folder
-        log_folder = LocalFileStorageManager.get_global_root(LocalFileStorageManager.LOGGING)
-        filesystem.ensure_folder_exists(log_folder)
+        # store new log name
+        self._std_file_handler_log_name = log_name
 
-        # generate log path
-        log_file = os.path.join(
-            log_folder,
-            "%s.log" % filesystem.create_valid_filename(log_name)
-        )
+        if log_name:
+            # set up a new handler
 
-        cls._std_file_handler = logging.handlers.RotatingFileHandler(
-            log_file,
-            maxBytes=4096*1024,
-            backupCount=5
-        )
+            # avoid cyclic references
+            from .util import filesystem
+
+            # set up logging root folder
+            filesystem.ensure_folder_exists(self.log_folder)
+
+            # generate log path
+            log_file = os.path.join(
+                self.log_folder,
+                "%s.log" % filesystem.create_valid_filename(log_name)
+            )
```

Comment:
When the engine starts, it only attempts to adjust the log file in case logging to file has yet started:
https://github.com/shotgunsoftware/tk-core/blob/improved_logging/python/tank/platform/engine.py#L2265

For the tank command, we start logging to file straight away:
https://github.com/shotgunsoftware/tk-core/blob/improved_logging/scripts/tank_cmd.py#L1419

JF is implementing the same sort of thing for desktop. Meaning we have logging into a single log file from a very early point in the execution. 

The check in start_engine ensures that when a DCC launches, logging kicks in at engine start if it's not already going.